cmd/clusterctl/client/cluster/objectgraph.go

```diff
@@ -473,6 +473,35 @@ func (o *objectGraph) Discovery(namespace string) error {
 	// Completes the graph by setting for each node the list of tenants the node belongs to.
 	o.setTenants()
 
+	// Filter and remove nodes in the graph that do not belong to cluster
+	if cluster != "" {
+		return o.filterCluster(cluster)
+	}
+
+	return nil
+}
+
+// filterCluster removes all objects but provided cluster and its dependents and soft-dependents
+func (o *objectGraph) filterCluster(cluster string) error {
+	for _, node := range o.getNodes() {
+		clusterTenant := false
+		for tenant, _ := range node.tenant {
+
+			// check if tenant is "Cluster" Kind and if the name matches with the provided cluster name
+			if tenant.identity.GroupVersionKind().GroupKind() == clusterv1.GroupVersion.WithKind("Cluster").GroupKind() {
+				if clusterTenant {
+					return fmt.Errorf("more than one cluster tenant identified for node - %s", node.identity.Name)
```

Comment:
should we throw this error only if one of the tenants is the cluster we are trying to move?

Example
3 clusters
We are trying to move cluster1
And we find an object than depends on cluster2 and cluster3. Do we need to throw an error? In this case ti should be fine, since we are not moving neither cluster 2 or cluster3.